Bottom: 03db07ce24f6bbdbdc29f0fb04b5eaf9f4322d49
Top:    a4e41811f849239e4049dccfb2304e407ae892b0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-22 23:52:20 +0000

TMP


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index eb8fbd1..0c48026 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -1513,6 +1513,15 @@ config_layer_group_button_state_update (void)
   for (g = 0; g < max_group; g++)
     for (i = 0; i < layer_groups.Number[g]; i++)
       {
+	/* Skip anything beyond the bottom silk / top silk layers residing
+	   at index max_copper_layer + 0 and max_copper_layer + 1. These are
+	   used to indicate the groups for bottom and top sides respectively,
+	   so form somewhat of a special case in our handling here (they have
+	   group buttons).
+	 */
+	if (layer_groups.Entries[g][i] >= max_copper_layer + 2)
+	  continue;
+
 /*			printf("layer %d in group %d\n", layer_groups.Entries[g][i], g +1); */
 	config_layer_group[layer_groups.Entries[g][i]] = g + 1;
 	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON
diff --git a/src/misc.c b/src/misc.c
index b01a5ae..ff78250 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2023,7 +2023,7 @@ MoveLayerToGroup (int layer, int group)
   int prev, i, j;
 
 #warning BUG NEXT LINE?
-  if (layer < 0 || layer > max_copper_layer + 1)
+  if (layer < 0 || layer >= max_copper_layer + 2)
     return -1;
   prev = GetLayerGroupNumberByNumber (layer);
   if ((layer == bottom_silk_layer
